### PR TITLE
Adding config repo module

### PR DESCRIPTION
**Pull request by ibnc**
Related to releasing the module work done in https://github.com/gocd/gocd/pull/5470

### CODE REVIEW

**Comment by ibnc** [comment]
When this PR is merged, the appropriate artifacts must be added to the installers pipeline

**Comment by arvindsv** [comment]
Let's have this be reviewed by someone other than the usual suspects (which are @ketan and me). Someone else should pick this up - and they'll need to understand a lot of the context around it to reproduce it locally to check - which helps spread knowledge and maybe they come up with improvements.

/cc @maheshp or @ibnc get to decide.

**Comment by ibnc** [comment]
@arvindsv 

That sounds reasonable to me. My initial thoughts are @GaneshSPatil or @marques-work ... Thoughts?

**Comment by maheshp** [comment]
@akshaydewan will be looking into this.

**Comment by akshaydewan** [comment]
The changes look good. 👍 

I didn't test it locally though, and I'm not sure how to go about doing that. Maybe we can just test with the `experimental` jar on the pipeline itself?

**Comment by arvindsv** [comment]
@akshaydewan I'd suggest trying to run the gradle task locally. If you need credentials, you should find them in the vault. Don't set `AUTO_RELEASE_TO_CENTRAL` to true!

If you need help / would like to pair, let me know.

**Comment by ibnc** [comment]
@arvindsv I'd be happy to test this out locally, but I don't have permissions. This will also fail locally because the artifacts have not been added to the installers pipelines on build.gocd.org

**Comment by ibnc** [comment]
I can confirm that `prepare_for_upload` is working fine. `fetch` isn't really testable until the artifact is added, and we add `plugin-infra:go-plugin-config-repo:assemble` to the task.

**Comment by arvindsv** [comment]
@ibnc Can you add the steps to reproduce that, so that @akshaydewan can take a look too?

**Comment by ibnc** [comment]
@arvindsv reproduce what exactly?

**Comment by ibnc** [comment]
Upload appears to be working. I removed the signing locally because I don't have the keys, so it eventually failed. I'm unsure if we need to add any rulesets to nexus

**Comment by arvindsv** [comment]
> @arvindsv reproduce what exactly?

@ibnc It's ok, don't bother.

Whenever possible, I try to make it easy for reviewers to review my PRs. In this case, I'd expect something like:

<hr/>

1. Download JAR from here.
2. Put it in this directory structure by running: `mkdir file/blah.jar`
3. Run: `./gradlew prepare-for-upload`

It should run successfully and you can see the JAR in staging at https://oss.sonatype.org/#stagingRepositories

<hr/>

In my opinion, this helps, since reviewers don't need to go through all that you just did to figure out the right order of things to do.

**Comment by arvindsv** [comment]
Examples: https://github.com/gocd/gocd/pull/5188 and https://github.com/gocd/gocd/pull/5022#issue-207758426

**Comment by ibnc** [comment]
@arvindsv gotcha.

I just added the artifact to the installers pipelines. Once this has been approved and merged, everything should be good to go as far as I know.

**Comment by ibnc** [comment]
After fixing an issue with fetching the artifacts, I tested the fetch locally. It's working just fine.

File: upload-to-maven [hunk 1]
```diff
@@ -37,6 +37,22 @@ def maven_release_version
   experimental? ? go_full_version : go_version
 end
 
+def description_for(artifact)
```
Comment (arvindsv) [inline upload-to-maven:40]: Could also be replaced by:

```ruby
details = {
  'go-plugin-api': {
    name: '...',
    description: '...'
  },
  'go-plugin-config-repo': {
    name: '...',
    description: '...'
  }
}
```

And then later using it as:

```ruby
details[artifact_name][:name]
```

**Pull request merged**